### DISCUSSION TITLE
Index reset without any information

### DISCUSSION THREAD

**Discussion by krivulcik** [ravendb/ravendb#16686] [Q&A] [answered]
We lost an index on one of our nodes (the cluster is running 5.4.104).

Our application suddenly started serving empty results. One of the nodes had the index in question in the "Paused until restart" state with 0 entries.

There is no trace of any reason in the Notification Center (in Studio), in the server logs (Log Level: Operations), or in Windows Event Viewer, so we can't determine what happened and how to prevent this from happening in the future.

We vaguely remember that something similar happened several months ago.

Is there a logging mechanism to log index-related entries? Logging at the Information level would be too excessive - the default buffer size of 100k entries in the Studio is filled in seconds. However, having an audit log for index-related activity (resetting, pausing, replacing, etc.) would be helpful.

Any suggestions about this would be appreciated.

**Comment by georgiosd** [comment]
Not exactly what you're asking for, but I recently added some of the indexes to the ASP.NET health check mechanism so I can include them in the monitoring.

**Comment by grisha-kotler** [comment, marked as answer]
There is the Audit Log.
It logs index creation and deletion.
https://ravendb.net/docs/article-page/5.4/csharp/server/security/audit-log/audit-log

**Reply by krivulcik** [reply]
Yes, this works well, thanks.
